Makefile.core.mk

```diff
@@ -536,12 +542,22 @@ lint-bundle: operator-sdk ## Run linters against OLM metadata bundle.
 lint-watches: ## Checks if the operator watches all resource kinds present in Helm charts.
 	@hack/lint-watches.sh
 
+lint-secrets: ## Checks whether any secrets are present in the repository.
+	@${GITLEAKS} detect --redact -v
+
 .PHONY: lint
-lint: lint-scripts lint-copyright-banner lint-go lint-yaml lint-helm lint-bundle lint-watches ## Run all linters.
+lint: lint-scripts lint-copyright-banner lint-go lint-yaml lint-helm lint-bundle lint-watches lint-secrets ## Run all linters.
 
 .PHONY: format
 format: format-go tidy-go ## Auto-format all code. This should be run before sending a PR.
 
+git-hook: gitleaks ## Installs gitleaks as a git pre-commit hook.
```

Comment:
Maybe add a not to the CONTRIBUTING.md file (or DEVELOPERS.md) about this?

e.g. see the second bullet here, which does the same: https://github.com/envoyproxy/envoy/blob/main/CONTRIBUTING.md#submitting-a-pr

Reply:
I added a few sentences to the `Contributing` section of the `README.md` -- I think we'll need to talk about `CONTRIBUTING.md` and `BUGS-AND-FEATURE-REQUESTS.md` as they currently are just copies of Istio's version... which is not really correct I think, we should probably have our own.